### DISCUSSION TITLE
Multiple custom 404 pages?

### DISCUSSION THREAD

**Discussion by tomasvorel2** [vercel/next.js#45082] [Help]
Hi everyone,

Is there a way to have multiple custom 404 pages? I have routes where I'm using a different layout and I want my 404 page to match that layout's design. I thought that I would just simply create a nested `404.tsx` page (e.g `/pages/some-page/404.tsx`) but this does not seem to work. Is this even possible? I'm using Next 12.3.2 so no app directory.

Thanks!

**Comment by jorshali** [comment]
You could do this yourself in the `/pages/404.tsx` page with something like the following:

```
export default function Custom404() {
  const router = useRouter();
  const [ fallback, setFallback ] = useState(false);

  React.useEffect(() => {
    if (!router.isReady) return;

    if (router.asPath.startsWith('/pages/some-page') {
      router.replace('/pages/some-page/404.tsx');
    } else {
      setFallback(true);
    }
  }, [router.isReady]);

  if (fallback) {
    // render general 404 content
  }

  return (
    <div></div>
  );
}
```

You could make this more dynamic if you wanted every sub-directory to have it's own 404 page.

A second option would be to customize the layout on `/pages/404.tsx` based on the `router.asPath` value.  Use similar `startsWith` logic to what you see above, but instead determine the layout based on the matching path.

**Comment by GabenGar** [comment]
Why do you have layout in a 404 page? The page exists as a clear marking of a dead end for crawlers and the only link it should have is the one to the home page of the site.
